### ISSUE TITLE
[java] Examples in documentation of MethodArgumentCanBeFinal do not trigger the rule

### ISSUE THREAD

**Issue by soyodream** [pmd/pmd#4779]
**Affects PMD Version:7.0.0-rc4** 

**Rule:MethodArgumentCouldBeFinal**
<https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#methodargumentcouldbefinal>

**Description:**
 According to the rule description, when defining a method with an empty body, the method's parameters should be marked as 'final.' However, the provided sample code, which meets the conditions described by the rule, is not flagged as a violation. This discrepancy indicates a false negative in the PMD analysis.

**Code Sample demonstrating the issue:**

```java
public class Foo{
    public void foo1 (String param) {  //should report violation

    }
}
```

**Expected outcome:**

PMD should report a violation at line 2, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
The rule ignores unused variables so as not to overlap with UnusedVariable. This should be clarified in the doc and examples, but is not a false negative.
